pkg/controller/controller.go

```diff
@@ -462,42 +480,53 @@ func convertCatalog(in *osb.CatalogResponse) ([]*v1alpha1.ServiceClass, error) {
 			if err != nil {
 				err = fmt.Errorf("Failed to marshal metadata\n%+v\n %v", svc.Metadata, err)
 				glog.Error(err)
-				return nil, err
+				return nil, nil, err
 			}
-			ret[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
+			serviceClasses[i].ExternalMetadata = &runtime.RawExtension{Raw: metadata}
 		}
 
-		ret[i].SetName(svc.Name)
+		serviceClasses[i].SetName(svc.Name)
+
+		// set up the plans using the ServiceClass Name
+		plans, err := convertServicePlans(svc.Plans, svc.Name)
```

Comment:
nit: pass `serviceClasses[i].Name` instead of `svc.Name` for cleanliness.

Reply:
I do not follow, what do you mean?

Reply:
2 lines above you take a value from `svc.Name` and set it to serviceClass.
`convertServicePlans` generates a plan which has a reference to serviceClass, right? So I would prefer to explicitly pass `serviceClasses[i].Name` here instead of `svc.Name` (even though the value is the same).

Imagine that you would change the code 2 lines-above to something like 
```go
serviceClasses[i].SetName(svc.Name + generateUuid())
```
in this case the name you pass into `convertServicePlans` would be out of sync. With my change it wouldn't.